networkx/classes/reportviews.py

```diff
@@ -187,7 +187,7 @@ def __iter__(self):
     def __getitem__(self, n):
         if isinstance(n, slice):
             raise nx.NetworkXError(
-                f"NodeView does not support slicing, try list(G.nodes)[{n.start}:{n.stop}]"
+                f"{type(self).__name__} does not support slicing, try list(G.nodes)[{n}]"
```

Comment:
I went with `{n.start}:{n.stop}` as IMO the target users for this error may not know about the slice object.

Reply:
Agreed - the ideal solution would be a string formatter that converts `slice` objects back to the colon-separated notation. I wasn't able to find any format spec for `slice` objects in Python.

I'm okay with sticking with the original reporting, but then the `step` should be included too, otherwise the error message will be wrong for slices that use steps:

```python
>>> G.nodes()[::2]
NetworkXError: NodeView does not support slicing, try list(G.nodes)[None:None]
```

Reply:
I went back to the `[start:stop:step]` style of reporting in 0c90e29 - LMK what you think!

Reply:
LGTM.